src/main/java/at/hannibal2/skyhanni/features/inventory/MaxPurseItems.kt

```diff
@@ -0,0 +1,90 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.PurseAPI
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.features.bazaar.BazaarApi
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.math.floor
+
+class MaxPurseItems {
+
+    private val orderPattern by RepoPattern.pattern("inventory.maxpurse.order", ".*§6(?<coins>[\\d.,]+) coins §7each.*")
+    private val instantPattern by RepoPattern.pattern(
+        "inventory.maxpurse.instant",
+        ".*Price per unit: §6(?<coins>[\\d.,]+) coins.*"
+    )
+    private val createOrderPattern by RepoPattern.pattern("inventory.maxpurse.createorder", "§aCreate Buy Order")
+    private val createInstantPattern by RepoPattern.pattern("inventory.maxpurse.createinstant", "§aBuy Instantly")
+    private var buyOrderPrice = -1f
+    private var instantBuyPrice = -1f
+
+    private fun getPrices() {
+        items@ for (item in Minecraft.getMinecraft().thePlayer.openContainer.inventory) {
+            createOrderPattern.matchMatcher(item?.displayName ?: continue) {
+                if (matches()) {
+                    lore@ for (info in item.getLore()) {
+                        orderPattern.matchMatcher(info) {
+                            if (!matches()) continue@lore
+                            buyOrderPrice = group("coins").replace(",", "").toFloat()
+                            // If we get to this point, we have the instant price because instant is earlier in the list of items
+                            // So we can return
+                            return
+                        }
+                    }
+                }
+            }
+            createInstantPattern.matchMatcher(item.displayName ?: continue) {
+                if (matches()) {
+                    lore@ for (info in item.getLore()) {
+                        instantPattern.matchMatcher(info) {
+                            if (!matches()) continue@lore
+                            instantBuyPrice = group("coins").replace(",", "").toFloat()
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
matchMatcher already ensures matcher.matches is true so this can reduce stuff like lore@ and items@ in your code so that it is less indented and more readable